sklearn/model_selection/tests/test_validation.py

```diff
@@ -1299,3 +1302,19 @@ def test_permutation_test_score_pandas():
         check_series = lambda x: isinstance(x, TargetType)
         clf = CheckingClassifier(check_X=check_df, check_y=check_series)
         permutation_test_score(clf, X_df, y_ser)
+
+
+def test_cross_val_score_labels_paramter():
```

Comment:
I would create an example where we know the outcome, and then check the outcome with different ``classes`` parameters. So maybe one where we have 3 classes in the data and then run the classifier with different settings of classes (like 3 or 4 or 5 classes) and without classes, and make sure the results are the one we expect.
You could write the cross-validation for-loop with ``KFold(...).split()`` and ``f1_score(average=None)`` to get the f1 scores per class and then check that the results of the macro-average are as expected.